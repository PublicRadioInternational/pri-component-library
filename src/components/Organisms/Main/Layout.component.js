/**
 * @file LayoutMain.component.js
 * Exports a main content section.
 */

import React from 'react';

import LayoutMain from './LayoutMain.component';
import LayoutMainList from './LayoutMainList.component';
import LayoutMainList2 from './LayoutMainList2.component';
import LayoutAsideLatest from './LayoutAsideLatest.component';
import LayoutAsideCallouts from './LayoutAsideCallouts.component';

/**
 * Component that renders the main content area.
 */
const Layout = () => (
  <LayoutMain>
    <LayoutMainList>
      <p>Main List</p>
    </LayoutMainList>
    <LayoutAsideLatest>
      <p>Latest Content</p>
    </LayoutAsideLatest>
    <LayoutAsideCallouts>
      <p>Callouts</p>
    </LayoutAsideCallouts>
    <LayoutMainList2>
      <p>Main List Two</p>
    </LayoutMainList2>
  </LayoutMain>
);

export default Layout;
