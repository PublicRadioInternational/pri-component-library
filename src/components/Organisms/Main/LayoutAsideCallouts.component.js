/**
 * @file LayoutMain.component.js
 * Exports a main content section.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Main.css';

import Aside from '../Content/Aside.component';

/**
 * Component that renders the main content area.
 */
const LayoutAsideCallouts = ({ children }) => (
  <Aside className={styles.callouts}>{children}</Aside>
);

LayoutAsideCallouts.propTypes = {
  children: PropTypes.node
};

LayoutAsideCallouts.defaultProps = {
  children: []
};

export default LayoutAsideCallouts;
