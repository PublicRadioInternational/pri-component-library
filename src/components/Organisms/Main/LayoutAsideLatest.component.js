/**
 * @file LayoutMain.component.js
 * Exports a main content section.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Main.css';

import Aside from '../Content/Aside.component';

/**
 * Component that renders the main content area.
 */
const LayoutAsideLatest = ({ children }) => (
  <Aside className={styles.latestContent}>{children}</Aside>
);

LayoutAsideLatest.propTypes = {
  children: PropTypes.node
};

LayoutAsideLatest.defaultProps = {
  children: []
};

export default LayoutAsideLatest;
