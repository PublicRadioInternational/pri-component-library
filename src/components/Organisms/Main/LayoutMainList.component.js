/**
 * @file LayoutMain.component.js
 * Exports a main list content section.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Main.css';

import Section from '../Content/Section.component';

/**
 * Component that renders the main content area.
 */
const LayoutSection = ({ children }) => (
  <Section className={styles.mainList}>{children}</Section>
);

LayoutSection.propTypes = {
  children: PropTypes.node
};

LayoutSection.defaultProps = {
  children: []
};

export default LayoutSection;
