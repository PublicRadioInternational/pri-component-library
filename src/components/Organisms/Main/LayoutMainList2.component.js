/**
 * @file LayoutMain.component.js
 * Exports a secondary content section.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Main.css';

import Section from '../Content/Section.component';

/**
 * Component that renders the main content area.
 */
const LayoutSection2 = ({ children }) => (
  <Section className={styles.mainList2}>{children}</Section>
);

LayoutSection2.propTypes = {
  children: PropTypes.node
};

LayoutSection2.defaultProps = {
  children: []
};

export default LayoutSection2;
