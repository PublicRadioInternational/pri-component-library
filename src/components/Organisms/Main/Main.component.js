/**
 * @file Main.component.js
 * Exports the site header.
 */

import React from 'react';
import styles from './Main.css';

/**
 * Component that renders the main content area.
 */
const Main = () => (
  <main className={styles.mainContainer}>
    <div className={styles.mainList}>
      <p>Main List</p>
    </div>
    <div className={styles.latestContent}>
      <p>Latest Content</p>
    </div>
    <div className={styles.callouts}>
      <p>Callouts</p>
    </div>
    <div className={styles.mainList2}>
      <p>Main List Two</p>
    </div>
  </main>
);

export default Main;
