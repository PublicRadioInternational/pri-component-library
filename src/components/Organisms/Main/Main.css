/* import colors... */
@value colors: "../../00_global/colors.css";
@value grayLighter, grayDark from colors;

/* import breakpoints */
@value header as bp-header, medium as bp-medium, maxWidth as bp-maxWidth from "../../00_global/breakpoints.css";

.drawerBg {
  background: rgba(0, 0, 0, 0.5);
  bottom: 0;
  top: 0;
  left: 0;
  position: fixed;
  width: 100%;
  height: 100%;
  z-index: 4000;
  opacity: 0.001;
  transform: translateZ(0);
  visibility: hidden;
}

/* Main Content */
.mainContainer {
  margin: 1rem 1rem 2rem;
  max-width: 1200px;
}

@media bp-maxWidth {
  .mainContainer {
    margin: 1rem auto 2rem;
  }
}

.mainList,
.mainList2,
.latestContent,
.callouts {
  margin-bottom: 2rem;
}

@media bp-medium {
  .mainContainer {
    display: -ms-grid;
    display: grid;
    -ms-grid-rows: auto;
    grid-template-rows: auto;
    -ms-grid-columns: 1fr 1fr;
    grid-template-columns: repeat(2, 16em);
  }

  .mainList {
    -ms-grid-column: 1;
    -ms-grid-column-span: 2;
    grid-column: 1 / span 2;
  }

  .latestContent {
    -ms-grid-column: 3;
    -ms-grid-column-span: 1;
    grid-column: 3 / span 1;
  }

  .callouts {
    -ms-grid-column: 3;
    -ms-grid-column-span: 1;
    -ms-grid-row: 2;
    grid-column: 3 / span 1;
  }

  .mainList2 {
    -ms-grid-column: 1;
    -ms-grid-row: 2;
    -ms-grid-row-span: 2;
    grid-row: row1-start / 2;
  }
}

@media bp-header {
  .mainContainer {
    -ms-grid-columns: 19% 54% 27%;
    grid-template-columns: 19% 54% 27%;
  }

  .mainList {
    -ms-grid-column: 2;
    -ms-grid-column-span: 2;
    grid-column: 2 / span 1;
    padding: 0 3%;
  }

  .latestContent {
    -ms-grid-column: 3;
    -ms-grid-column-span: 1;
    grid-column: 3 / span 1;
  }

  .callouts {
    -ms-grid-column: 1;
    -ms-grid-column-span: 1;
    -ms-grid-row: 1;
    -ms-grid-row-span: 1;
    grid-column: 1 / span 1;
    grid-row: row1-start / 1;
  }

  .mainList2 {
    -ms-grid-column: 2;
    -ms-grid-column-span: 2;
    -ms-grid-row: 2;
    grid-row: row1-start / 2;
    padding: 0 3%;
  }
}
