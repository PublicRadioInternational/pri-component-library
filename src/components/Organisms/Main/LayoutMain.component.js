/**
 * @file LayoutMain.component.js
 * Exports a main content section.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Main.css';

import Main from './Main.component';

/**
 * Component that renders the main content area.
 */
const LayoutMain = ({ children }) => (
  <Main className={styles.mainContainer}>{children}</Main>
);

LayoutMain.propTypes = {
  children: PropTypes.node
};

LayoutMain.defaultProps = {
  children: []
};

export default LayoutMain;
