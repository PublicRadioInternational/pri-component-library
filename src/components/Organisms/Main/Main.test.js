/**
 * @file Main.test.js
 * Contains tests for Main.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Main from './Main.component';
import Layout from './Layout.component';

describe('<Main />', () => {
  it('Matches the Main content area snapshot', () => {
    const component = renderer.create(<Main />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('<Layout />', () => {
  it('Matches the Layout content area snapshot', () => {
    const component = renderer.create(<Layout />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
