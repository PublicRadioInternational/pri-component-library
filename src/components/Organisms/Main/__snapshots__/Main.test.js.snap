// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Matches the Main content area snapshot 1`] = `
<main
  className="mainContainer"
>
  <div
    className="mainList"
  >
    <p>
      Main List
    </p>
  </div>
  <div
    className="latestContent"
  >
    <p>
      Latest Content
    </p>
  </div>
  <div
    className="callouts"
  >
    <p>
      Callouts
    </p>
  </div>
  <div
    className="mainList2"
  >
    <p>
      Main List Two
    </p>
  </div>
</main>
`;
